### DISCUSSION TITLE
Is make install still supported to install the built ACE/TAO?

### DISCUSSION THREAD

**Discussion by pierre-rouleau** [DOCGroup/ACE_TAO#2266] [Q&A]
I followed the instructions in the [Building and Installing ACE on UNIX](https://htmlpreview.github.io/?https://github.com/DOCGroup/ACE_TAO/blob/master/ACE/ACE-INSTALL.html#unix) , **Using the Traditional ACE/GNU Configuration**  to build ACE/TAO 7.0.1 on Rocky Linux 8.10.

After setting the appropriate environment variables and files, as described in the document, I was able to build ACE: the `make` command **succeeded**.

I also had the `INSTALL_PREFIX` and `LD_LIBRARY_PATH` environment variables set.  
But then running  `make install` **failed**  with:
``` shell
make[1]: Entering directory '/local_home/rouleau/my/dvpub/ace-tao-7.0.11/ACE_wrappers/ace'

GNUmakefile: /local_home/rouleau/my/dvpub/ace-tao-7.0.11/ACE_wrappers/ace/GNUmakefile.ACE MAKEFLAGS=w

perl -ne'if (/^#END MPC-Generated Install Info/) {exit 0}' \
  -e'elsif (/^#BEGIN MPC-Generated Install Info/) {$ok=1}' \
  -e'elsif ($ok && s/^#//) {print}' GNUmakefile.ACE | \
/usr/share/mpc/prj_install.pl -i -s lib_output,header_files,template_files,inline_files,installdata_files \
  -b header_files=include/ace -b template_files=include/ace -b inline_files=include/ace -o lib_output=lib -x installdata_files -b installdata_files=share/ace   \
  /usr/local
ERROR: Unable to copy Abstract_Timer_Queue.cpp to /usr/local/include/ace/./Abstract_Timer_Queue.cpp
make[1]: *** [GNUmakefile.ACE:529: install] Error 1
make[1]: Leaving directory '/local_home/rouleau/my/dvpub/ace-tao-7.0.11/ACE_wrappers/ace'
```

**My question:**   On ACE/TAO 7.0.1 (which I attempted to build), is this `make; make install` way of building ACE (on Linux) described in the document I referenced above ,expected to succeed (and then I'd need to find the access problem and resolve them)?  Or is this an old/obsolete/non-supported build method and I should look for something else, like a `rpmbuild`?

**Comment by mitza-oci** [comment]
The `make install` command needs write permissions to the location where it's going to be installed.

**Reply by pierre-rouleau** [reply]
Thanks for the reply. 
The instructions should probably be updated to  change the `make install` to `sudo -E make install` then.
I could update the instructions if that would be considered useful.

**Reply by mitza-oci** [reply]
It is deliberately not written that way.  Many use cases involve installing to a location that's already writable by the current user.

Instead of having a user run `sudo` when it may be required, the instructions could mention the fact that the `INSTALL_PREFIX` location must be writable.

**Reply by pierre-rouleau** [reply]
It was obvious to me that the /usr/include might not be writable and that it might be the reason for the failure, and it's why I asked the question in the first place.

For the systems I built in the past the `sudo make install` would be working fine but that does not carry the required environment variables so `sudo -E` is required (and user must have access to it).  Not all users have access to a system where they can use `sudo -E`.

What was the reason behind not mentioning the potential need to run as sudo the `make install`?  Is this not kind of universal in various Unixes?  sudo is not installed by default on some BSD but is available.  

Perhaps writing that INSTALL_PREFIX identified directory must be user writable *or* use `sudo -E` would cover all use cases?

**Reply by pierre-rouleau** [reply]
Well... I guess updating the docs is not something I should consider.

**Reply by mitza-oci** [reply]
I think that specifying the mechanism by which the user may elevating privileges to gain write access is beyond the scope of the document.  

The document could mention that write permissions are required and that different systems have different ways of gaining them (if the user doesn't already have them).
